hy/compiler.py

```diff
@@ -1709,10 +1722,22 @@ def hy_compile(tree, module_name, root=ast.Module, get_expr=False):
 
     tree = wrap_value(tree)
     if not isinstance(tree, HyObject):
-        raise HyCompileError("`tree` must be a HyObject or capable of "
-                             "being promoted to one")
-
-    compiler = HyASTCompiler(module_name)
+        raise TypeError("`tree` must be a HyObject or capable of "
+                        "being promoted to one")
+
+    try:
+        mod_loader = pkgutil.get_loader(module_name)
+        if filename is None:
+            filename = mod_loader.get_filename()
+        # TODO: Lazy load this.
```

Comment:
I should ask: are TODO comments like this meant for yourself or for future readers and editors of the codebase?

Reply:
Heh, good question: sometimes one of those, sometimes a couple, sometimes all!  
This one is for editors, though.

Reply:
I see. I think an idea about how the code should change in the future, like that, is better as a GitHub issue than a comment. Otherwise we're kind of tracking things we want to do in two different places.